File: GGAndroid_2.0/GGPay/pay-alipay/src/main/java/io/ganguo/pay/alipay/AlipayMethod.java
package io.ganguo.pay.alipay;

import android.app.Activity;

import io.ganguo.pay.core.GGPay;
import io.ganguo.pay.core.IPayService;
import io.ganguo.pay.core.PayCallback;

/**
 * Created by Roger on 07/07/2017.
 */

public class AlipayMethod implements IAlipayMethod {
    @Override
    public void pay(Activity activity, PayCallback payCallback, String orderInfo) {
        GGPay.newService(new AlipayServiceProvider(activity, payCallback, orderInfo)).pay();
    }

    @Override
    public IPayService asPayService(Activity activity, PayCallback payCallback, String orderInfo) {
        return GGPay.newService(new AlipayServiceProvider(activity, payCallback, orderInfo));
    }
}

File: GGAndroid_3.0/core/service/pay/pay-alipay/src/main/java/io/ganguo/pay/alipay/AliPayMethod.kt
package io.ganguo.pay.alipay

import android.app.Activity

/**
 * Created by Roger on 07/07/2017.
 * 支付宝支付
 * 支付结果处理在[PayResultHandler]
 */
class AliPayMethod : AAliPayMethod() {

    /**
     * AliPayService 创建方法
     * @param param 支付订单请求数据
     * @return
     */
    override fun newPayProvider(activity: Activity, param: String?): AliPayProvider {
        return AliPayProvider(activity, param!!)
    }
}
